packages/rocket-batch-file-process-aws-infrastructure/src/file-to-line-event-lambda/index.ts

```diff
@@ -0,0 +1,90 @@
+import { S3Event } from 'aws-lambda'
+import { DynamoDB, S3 } from 'aws-sdk'
+
+const CSV_FILE = 'csv'
+const JSONL_FILE = 'jsonl'
+
+const s3 = new S3()
+const ddb = new DynamoDB.DocumentClient()
+const { v4: uuidv4 } = require('uuid')
+
+const ENTITY_ID_NAME = process.env.ENTITY_ID!
+
+export const handler = async (event: S3Event): Promise<void> => {
+  const {
+    bucket: { name },
+    object: { key },
+  } = event.Records[0].s3
+
+  const params = {
+    Bucket: name,
+    Key: key,
+  }
+
+  const data = await s3.getObject(params).promise()
+  if (data.Body) {
+    const fileExtension = key.split('.').pop()
+    const rows = data.Body.toString().split('\n')
+
+    if (fileExtension == CSV_FILE) {
+      await csvRowsToEvents(rows)
+    } else if (fileExtension == JSONL_FILE) {
+      await jsonlRowsToEvents(rows)
+    } else {
+      throw Error('File Format is not supported')
+    }
+  }
+}
+
+async function csvRowsToEvents(rows: string[]): Promise<void> {
+  const headers: string[] = rows.shift()!.split(',')
+  const indexOfEntityId = headers.indexOf(ENTITY_ID_NAME)
+
+  for (let rowNumber = 0; rowNumber < rows.length; rowNumber++) {
+    const rowValues: string[] = rows[rowNumber].split(',')
+    const entityIdValue = rowValues[indexOfEntityId]
+
+    let index = 0
+    const boosterEvent = {} as any
+
+    for (let headerNumber = 0; headerNumber < headers.length; headerNumber++) {
+      boosterEvent[headers[headerNumber]] = rowValues[index]
+      index++
+    }
+
+    await saveEvent(entityIdValue, boosterEvent)
+  }
+}
+
+async function jsonlRowsToEvents(rows: string[]): Promise<void> {
+  rows.shift()!
+
+  for (let rowNumber = 0; rowNumber < rows.length; rowNumber++) {
+    const boosterEvent = JSON.parse(rows[rowNumber])
+    const entityIdValue = boosterEvent[ENTITY_ID_NAME]
+    await saveEvent(entityIdValue, boosterEvent)
+  }
+}
+
+async function saveEvent(entityIdValue: string, boosterEvent: any): Promise<void> {
+  const params = {
+    TableName: process.env.EVENT_STORE_NAME!,
```

Comment:
I wonder if we could have a better way of passing parameters than environment variables. It looks a bit fragile...

Reply:
The problem is we don't control the input params of the lambda as they are S3Events triggered. So the only I can think of passing them as env variables of those lambdas. 

We could setup a DB and save those as well but not sure if we are gaining much more.